Feat/host changer

wildintellect opened the pull request:
Before you submit a pull request, please fill in the following:

**Related Issue(s):**
#3 

**Description:**
Added a CLI optional arg to both create_item and create_collection `--host` this specifies the Provider HOST to be picked from a dictionary of options in the constants.py

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

wildintellect committed:
feat: add option to specificy host

Host is now a cli argument on collections to allow the same recipe to be used on multiple implementations

wildintellect committed:
feat: Add host selection to Items too

wildintellect committed:
Merge branch 'feat/collection' into feat/host-changer

wildintellect committed:
typo: Singer/Sinerg

wildintellect committed:
chore: Update Changelog

wildintellect committed:
fix:#1 corrections

Added missing Collection on Items, added links to Collection

chuckwondo commented on src/stactools/cop_dem/constants.py:
```diff
@@ -45,16 +53,8 @@
- by the Copernicus Programme. Note that in both cases ocean areas do not have tiles,
- there one can assume height values equal to zero.
- Data is provided as Cloud Optimized GeoTIFFs and comes from Copernicus DEM 2021 release.'''
+COP_DEM_DESCRIPTION = '''The Copernicus DEM is a Digital Surface Model (DSM) which represents the surface of the Earth including buildings, infrastructure and vegetation. We provide two instances of Copernicus DEM named GLO-30 Public and GLO-90. GLO-90 provides worldwide coverage at 90 meters. GLO-30 Public provides limited worldwide coverage at 30 meters because a small subset of tiles covering  ...
```
```suggestion
COP_DEM_DESCRIPTION = '''The Copernicus DEM is a Digital Surface Model (DSM) that represents the surface of the Earth including buildings, infrastructure and vegetation. We provide two instances of Copernicus DEM, named GLO-30 Public and GLO-90. GLO-90 provides worldwide coverage at 90 meters. GLO-30 Public provides limited worldwide coverage at 30 meters because a small subset of tiles covering specific countries are not yet released to the public by the Copernicus Programme. Note that in both cases, ocean areas do not have tiles, where one can assume height values equal to zero. Data is provided as Cloud Optimized GeoTIFFs and comes from Copernicus DEM 2021 release.'''  # noqa: E501
```

wildintellect replied:
Was copied from AWS documents supplied by the provider, not going to change.

chuckwondo commented on src/stactools/cop_dem/commands.py:
```diff
@@ -19,8 +19,10 @@ def cop_dem():
                   help="Validate the item before saving")
-    def create_item_command(source, destination, validate):
-        item = stac.create_item(source)
+    @click.option("--host", default=None, help="Set PROVIDER HOST")
```
Do you want to limit accepted values to the keys in `co.COM_DEM_HOST`? If so, consider this (with required import as well):

```suggestion
    @click.option("--host",
                  type=click.Choice(co.COP_DEM_HOST, case_sensitive=False),
                  default=None,
                  help="Set PROVIDER HOST")
```

However, offhand, I don't know if `click` allows such choice options to be optional (i.e., with `default=None`). In other words, I don't know if specifying choices forces the option to be required, such that any value supplied as the default must be one of the choices (and thus cannot be None, unless perhaps None is one of the choices).

wildintellect replied:
We can test that idea.

wildintellect replied:
Also going to note that `click.Choice` wants a list so we need to get the keys from the Dict of possible hosts, not the dict.

chuckwondo commented on src/stactools/cop_dem/commands.py:
```diff
@@ -38,15 +40,16 @@ def create_item_command(source, destination, validate):
     @click.option("--validate/--no-validate",
                   default=True,
                   help="Validate the item before saving")
+    @click.option("--host", default=None, help="Set PROVIDER HOST")
```
Same comment as above regarding restricting valid host values.

chuckwondo reviewed (approved):
Looks good to me. I only have a couple of minor suggestions that you might consider, but feel free to ignore.

wildintellect committed:
feat: Improve cli handling of options

host is optional but when supplied comes from a list, product also is limited to a specific list

wildintellect committed:
chore: Validate Host on items too

chuckwondo reviewed (approved):
Looks good for now.
